src/plugins/expressions/common/service/expressions_services.ts

```diff
@@ -256,6 +257,34 @@ export class ExpressionsService {
     return fork;
   };
 
+  /**
+   * Migrates expression AST to the latest version. This function should be called whenever expression is read from any place where expression could be from previous kibana version.
+   * @param state old version of expression AST
+   * @param version kibana version of expression AST passed in
+   */
+  public readonly migrate = (state: ExpressionAstExpression) => {
+    return this.executor.migrate(state);
+  };
```

Comment:
not important, but one thought:

Do you think it's worth optionally handling string parsing as a convenience?

So if you call with an expression string, we return an expression string, or if you call with an AST, return with an AST?
```ts
migrate('foo | bar'); // => 'foo | migrated'
migrate({ ...ast }); // => { ...astMigrated }
```

Reply:
i would leave this out at first, we can always add that later if we find it helpful